modules/metadata/src/main/scala/com/snowplowanalytics/snowplow/scalatracker/metadata/package.scala

```diff
@@ -0,0 +1,31 @@
+package com.snowplowanalytics.snowplow.scalatracker
+
+import cats._
+import cats.implicits._
+import cats.effect.Sync
+
+package object metadata {
+
+  implicit class TrackerMetadataOps[F[_]: Applicative](val tracker: Tracker[F]) {
+
+    /**
+     * Adds EC2 context to each sent event
+     * Blocks event queue until either context resolved or timed out
+     */
+    def enableEc2Context[G[_]: Sync]: G[Tracker[F]] =
```

Comment:
Let me know if you are okay with this solution.

Reply:
I like this solution much more than previous one - it basically says "you can enable this feature, but be aware that it is not free", while previous one was very implicit about it.